src/oid_array.rs

```diff
@@ -0,0 +1,55 @@
+//! Bindings to libgit2's raw git_strarray type
+
+use std::ops::Deref;
+
+use oid::Oid;
+use raw;
+use util::Binding;
+use std::slice;
+use std::mem;
+
+/// An oid array structure used by libgit2
+///
+/// Some apis return arrays of oids which originate from libgit2. This
+/// wrapper type behaves a little like `Vec<&Oid>` but does so without copying
+/// the underlying Oids until necessary.
+pub struct OidArray {
+    raw: raw::git_oidarray,
+}
+
+impl OidArray {
+    /// Returns None if i is out of bounds.
+    pub fn get(&self, i: usize) -> Option<Oid> {
```

Comment:
I think this method is also present on slices, so perhaps it could be deferred in favor of that?